CustomChange transactional behaviour different since 4.28.0

### Search first

- [X] I searched and no similar issues were found

### Description

We are using PostgreSQL and are a long user of liquibase. Since the latest release `4.28.0` we observed an issue which we can only fix by rolling back to `4.27`.

With our existing ChangeSets that start our db state from scratch we now get the following error:

`Caused by: org.postgresql.util.PSQLException: ERROR: cannot ALTER TABLE "users" because it is being used by active queries in this session`

This happens on multiple places if we comment out certain problematic lines. It appears transactionality behaviour is different ? We pinpointed it is always related to a customChange. For example the following:

    <changeSet id="someid" author="myself
        <customChange class="<class implementing CustomTaskChange"></customChange>
        <addUniqueConstraint tableName="users" columnNames="mail_id, home_id"/>
    </changeSet>

In case the customChangeset uses the same table we get the error. It seems that previously the connection was committed/closed right after the customtaskchange and now the same one is used leading to the error we are getting.

We might fix this by revisting all the places we have a problem, however what is the correct and expected behaviour here, as this clearly changed since the latest version.

### Steps To Reproduce

See example in description

### Expected/Desired Behavior

The behaviour is as-is with previous versions.

### Liquibase Version

4.28.0

### Database Vendor & Version

PostgreSQL 14

### Liquibase Integration

gradle, spring boot, java

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

_No response_

### Additional Context

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

--- a/liquibase-standard/src/main/java/liquibase/executor/jvm/JdbcExecutor.java
+++ b/liquibase-standard/src/main/java/liquibase/executor/jvm/JdbcExecutor.java
@@ -75,7 +75,7 @@ public class JdbcExecutor extends AbstractExecutor {
                 throw new DatabaseException("Cannot execute commands against an offline database");
             }
             stmt = ((JdbcConnection) con).getUnderlyingConnection().createStatement();
-            if (Boolean.TRUE.equals(SqlConfiguration.ALWAYS_SET_FETCH_SIZE.getCurrentValue())) {
+            if (database instanceof OracleDatabase && Boolean.TRUE.equals(SqlConfiguration.ALWAYS_SET_FETCH_SIZE.getCurrentValue())) {
                 stmt.setFetchSize(database.getFetchSize());
             }
             Statement stmtToUse = stmt;

